File: Leetcode/python/3Sum Closest.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
        sort(nums.begin(), nums.end());
        int mn = INT_MAX;
        int res;
        int l = nums.size();

        for(int i=0; i<l-2; i++){
        	int left = i+1;
        	int right = l-1;

        	while (left < right){
        		int sum_ = nums[i] + nums[left] + nums[right];
        		int difference = abs(target - sum_);
        		if (difference < mn){
        			mn = difference;
        			res = sum_;
        		}
        		if(target - sum_ > 0)
        			left ++;
        		else if(target - sum_<0)
        			right--;
        		else
        			return res;
        	}
        }

        return res;

    }
};

File: Leetcode/python/3Sum Closest.py
class Solution:
    def threeSumClosest(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        nums.sort()
        l = len(nums)
        mn = 2000000000

        for i in range(l-2):
            left = i+1
            right = l-1
            while left < right:
                sum_3 = nums[left] + nums[right] + nums[i]
                if abs(target - sum_3) < mn:
                    mn = abs(target - sum_3)
                    res = sum_3
                mn = min(mn, abs(target - sum_3))
                if sum_3 < target:
                    left += 1
                elif sum_3 > target:
                    right -= 1
                else:
                	return sum_3
        return res

s = Solution()
nums = [0,1,2]
target = 3

print(s.threeSumClosest(nums, target))